segmentation/utils/synthetic_spatial_datagen.py

```diff
@@ -136,37 +134,91 @@ def point_init_dist_matrix(size_img=(1024, 1024), num_A=100, num_B=100, num_C=10
     # because we're passing these into skimage.measure.label, it is important
     # that we convert these to integers beforehand
     # since label only takes a binary matrix
-    # we pass the result through the unique function to eliminate any possibility of duplicate points
-    # appearing within any of these arrays
-    a_points = np.unique(np.random.multivariate_normal(a_mean, a_cov, num_A).astype(np.int16), axis=0)
-    b_points = np.unique(np.random.multivariate_normal(b_mean, b_cov, num_B).astype(np.int16), axis=0)
-    c_points = np.unique(np.random.multivariate_normal(c_mean, c_cov, num_C).astype(np.int16), axis=0)
+    a_points = np.random.multivariate_normal(a_mean, a_cov, num_A).astype(np.int16)
+    b_points = np.random.multivariate_normal(b_mean, b_cov, num_B).astype(np.int16)
+    c_points = np.random.multivariate_normal(c_mean, c_cov, num_C).astype(np.int16)
+
+    # combine the points together into one list
+    total_points = np.concatenate((a_points, b_points, c_points), axis=0)
+
+    # remove points with negative values since they're out of range
+    total_points = total_points[np.logical_and(sample_arr[:, 0] >= 0, sample_arr[:, 1] >= 0), :]
+
+    # remove points with values greater than the size_img dimensions since they're out of range
+    total_points = total_points[np.logical_and(sample_arr[:, 0] < size_img[0], sample_arr[:, 1] < size_img[1]), :]
 
     # this ensures that we only keep the points that are not duplicate across different cell types
-    points, counts = np.unique(np.concatenate((a_points, b_points, c_points), axis=0), axis=0, return_counts=True)
-    non_dup_points = points[counts == 1]
+    non_dup_points, non_dup_counts = np.unique(total_points, axis=0, return_counts=True)
+    total_points = non_dup_points[non_dup_counts == 1]
+
+    return total_points
+
+
+def point_init_dist_matrix(size_img=(1024, 1024), num_A=100, num_B=100, num_C=100,
```

Comment:
This function doesn't actually create a distance matrix anymore, so we should update the name to reflect its functionality. Something like generate_synthetic_label_map or whatever you think is appropriate.